plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveFailureRecovery.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import io.trino.Session;
+import io.trino.testing.AbstractTestFailureRecovery;
+import io.trino.testing.QueryRunner;
+import io.trino.tpch.TpchTable;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestHiveFailureRecovery
```

Comment:
Actually, I think we should force the Hive test which write table data to fail if failure occurs. I think that engine should consult connector if making a query retry is feasible (i.e does connector support proper commit protocol, or retry may result in bad data written to the table).

Reply:
Yeah, ideally connectors should explicitly tell the engine whether atomic writes are supported. Perhaps we may want to extend the connector interface to allow querying whether a DML operation on a specific table guaranteed to be atomic?

Reply:
I added that as part of https://github.com/trinodb/trino/pull/10064.
Also it is not only about writes. Connector must explicitly say that it support retries on reads too. Probably for query-level retries it should always work. But for tasks level retries not necessarily. IIUC for task level retries we assume that you can "reread" as Split. This is not true for every connector.

Reply:
> IIUC for task level retries we assume that you can "reread" as Split. This is not true for every connector.

Hmm... that's interesting. I haven't thought about it. Do you know what connectors don't support "re-reading" a split?